src/Microsoft.OData.Core/SelectedPropertiesNode.cs

```diff
@@ -469,6 +503,44 @@ private static IEnumerable<string> GetPossibleMatchesForSelectedOperation(IEdmOp
             yield return qualifiedContainerName + operationNameWithParameters;
         }
 
+        private bool IsValidExpandToken(string item)
+        {
+            // Check #1: Expand token must contain '(', must be related to navigation property.
+            int idxRP = item.IndexOf('(');
+            if (idxRP == -1
+                || (this.structuredType != null &&
+                    !(this.structuredType.FindProperty(item.Substring(0, idxRP)) is IEdmNavigationProperty)
+                    )
+                )
+            {
+                // selected item does not contain '(',
+                // or is not navigation property (for example when an Action is selected).
+                return false;
+            }
+
+            // Check #2: paretheses must be balanced.

```

Comment:
>// Check #2: paretheses must be balanced. [](start = 12, length = 41)

I'm not sure this check is necessary; I don't think we have any valid entries in the selectlist that allow unbalanced parens.  If we do need this check, then we should probably also verify that the last character was a closing paren?

Reply:
Agree. That is just a safeguard/double-check that is meaningful in the context of this function, but can be omitted for sake of optimization.